ec/provider_config.go

```diff
@@ -58,6 +74,18 @@ func configureAPI(d *schema.ResourceData) (interface{}, error) {
 		Timeout:         timeout,
 		UserAgent:       userAgent(Version),
 	})
+
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to create EC client",
+			Detail:   "Unable to authenticate user for EC client",
+		})
+
+		return nil, diags
+	}
+
+	return api, diags
```

Comment:
We can return `nil` directly here as all errors are fatal and we are not collecting a series of diagnostics yet.
```suggestion
	return api, nil
```